scripts/clang_tidy_wrapper.sh

```diff
@@ -10,13 +10,13 @@ shift  # Remove the file from $@
 file_relative_to_source=${file//$cmake_source_dir/}             # Remove the source dir from $file
 file_relative_to_source=${file_relative_to_source/..\/src/src}  # Remove `../src`, which gets added by Ninja
 
-if grep "$file_relative_to_source" $cmake_source_dir/.clang-tidy-ignore > /dev/null; then
+if grep -- "$file_relative_to_source" $cmake_source_dir/.clang-tidy-ignore > /dev/null; then
 	echo "clang-tidy: Ignoring $file_relative_to_source"
 	exit 0
 else
 	if [ -x /usr/local/bin/brew ]; then
```

Comment:
This line won't work for ARM Macs btw. I did not change it in #2489 since I we were not able to run recent clang-tidy versions anyways.

Reply:
I never considered clang-tidy and macOS. Shall we add a check for macOS and fail? Or revert this change (I can't remember what the reason was ... do you?)

Reply:
No idea about the grep change tbh.
IMO there's no reason for not allowing clang tidy on macOS, as it should not yield different warnings than the Linux version. We could check for the macOs or do sth like `if ! command -v  brew &> /dev/null`.

Reply:
Well, we need `--` for search terms that start with a dash. But since we grep for file names, I cannot think of a case where we  might have a search pattern starting with a dash here.

So I'll remove the `--` again.

Reply:
Still, we call `clang-tidy $file` instead of `$(brew --prefix llvm)/bin/clang-tidy $file` for ARM Macs, right?

Reply:
Sorry, I always thought it's all about the `--`. Misunderstood you.

Stackoverflow recommends:
```
which -s brew
if [[ $? != 0 ]] ; then
```

Reply:
Guess that's nice and short, no strong opinion for either `uname == "Darwin"` (used in other scripts) or `which -s brew`.

Reply:
Using `which brew > /dev/null` now. Works on macOS and Ubuntu. `-s` is not available on Linux.